velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -226,23 +369,80 @@ void WindowFuzzer::go() {
         makeFunctionCall(signature.name, argNames, false, false, ignoreNulls);
 
-    std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
-    // 50% chance without order-by clause.
-    if (vectorFuzzer_.coinToss(0.5)) {
-      sortingKeysAndOrders =
-          generateSortingKeysAndOrders("s", argNames, argTypes);
-    }
+    core::WindowNode::WindowType frameType;
+    core::WindowNode::BoundType startBoundType;
+    core::WindowNode::BoundType endBoundType;
+    std::tie(frameType, startBoundType, endBoundType) = generateFrameClause();
+
+    auto isKBoundFrame = [&](core::WindowNode::BoundType boundType) {
+      return (
+          boundType == core::WindowNode::BoundType::kPreceding ||
+          boundType == core::WindowNode::BoundType::kFollowing);
+    };
+
+    bool isKRangeFrame = frameType == core::WindowNode::WindowType::kRange &&
+        (isKBoundFrame(startBoundType) || isKBoundFrame(endBoundType));
+    auto hasRowNumberKey =
+        requireSortedInput || frameType == core::WindowNode::WindowType::kRows;
+
     const auto partitionKeys = generateSortingKeys("p", argNames, argTypes);
-    const auto [frameClause, isRowsFrame] = generateFrameClause();
-    const auto input = generateInputDataWithRowNumber(
-        argNames, argTypes, partitionKeys, signature);
+    std::vector<SortingKeyAndOrder> sortingKeysAndOrders;
     // If the function is order-dependent or uses "rows" frame, sort all input
     // rows by row_number additionally.
-    if (requireSortedInput || isRowsFrame) {
+    if (hasRowNumberKey) {
       sortingKeysAndOrders.emplace_back("row_number", core::kAscNullsLast);
       ++stats_.numSortedInputs;
     }
 
+    TypeKind orderByTypeKind;
+    // kRange frames need only one order by key. This should be row_number for
+    // functions that are order dependent.
+    if (isKRangeFrame) {
+      sortingKeysAndOrders = generateSortingKeysAndOrders(
+          "s", argNames, argTypes, hasRowNumberKey, isKRangeFrame);
+      orderByTypeKind = argTypes[argTypes.size() - 1]->kind();
+    } else if (vectorFuzzer_.coinToss(0.5)) {
+      // 50% chance without order-by clause.
+      sortingKeysAndOrders =
+          generateSortingKeysAndOrders("s", argNames, argTypes);
+    }
+
+    auto input = generateInputDataForWindowFuzzer(
+        argNames, argTypes, partitionKeys, signature, hasRowNumberKey);
+    if (isKRangeFrame) {
+      // Catch possible type overflow errors when generating offset columns.
+      try {
+        VELOX_USER_CHECK(
+            sortingKeysAndOrders.size() == 1,
+            "Window with k PRECEDING/FOLLOWING frame bounds should have a single ORDER-BY key");
+        auto orderByKey = sortingKeysAndOrders[0];
+        VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
+            addOffsetColumnsToInput,
+            orderByTypeKind,
+            input,
+            startBoundType,
+            endBoundType,
+            orderByKey);
+      } catch (VeloxUserError& e) {
+        continue;
```

Comment:
Could we add a log before continuing that `fmt::format("This iteration is not valid due to exception from addKRangeOffsetColumnsToInput: {}", e.message())`?